adriene-orange/react-sandbox
# Simple MirageJS To Do List POC

This adds MirageJS as a mock server to run in the client for a mock workflow/cypress tests
or as a part of unit/integration tests.

How to test this:
- Pull down this branch
- npm i
- npm start // to run client
This setup is set to load 3 TODO tasks from the mock server
Adding, deleting and updating tasks should work
Because this runs in the browser, refreshing the page causes the DB to restart

- npm test // to run tests
- The `App` test renders the entire app in memory, runs the mock server and then interacts with the DOM like a user.


Not related changes (sorry)
- Added some linting because it was driving me crazy
- Added some more states to the client to add interesting test scenarios
